docs/usage/validation_errors.md

```diff
@@ -3,251 +3,1470 @@ with some suggestions on how to fix them.
 
 ## `arguments_type`
 
+This error is raised when arguments passed to a function are not
+a tuple, list or a dictionary.
 
 ## `assertion_error`
 
+This error is raised when there is a failing assertion during the validation.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel, field_validator
+
+
+class Model(BaseModel):
+    x: int
+
+    @field_validator('x')
+    @classmethod
+    def force_x_positive(cls, v):
+        assert v > 0
+        return v
+
+Model(x=-1)
+```
 
 ## `bool`
 
 
 ## `bool_parsing`
 
+This error is raised when the value is an invalid boolean string.
+
+```py test="skip" lint="skip" upgrade="skip"
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    x: bool
+
+print(Model(x='test'))
```

Comment:
```suggestion
print(Model(x='true'))  # OK
print(Model(x='test'))
```
I think it's maybe worth making clear via this kind of simple example here that some strings _are_ acceptable